packages/workspace/src/command-line/run-tasks/run-command.ts

```diff
@@ -0,0 +1,205 @@
+import { DependencyGraph, NxJson, ProjectNode } from '../shared';
+import {
+  AffectedEventType,
+  Task,
+  TaskCompleteEvent,
+  TasksRunner
+} from '../../tasks-runner/tasks-runner';
+import { defaultTasksRunner } from '../../tasks-runner/default-tasks-runner';
+import { isRelativePath } from '../../utils/fileutils';
+import { join } from 'path';
+import { appRootPath } from '../../utils/app-root';
+import { DefaultReporter, ReporterArgs } from './default-reporter';
+import {
+  Arguments,
+  Environment,
+  projectHasTargetAndConfiguration
+} from './utils';
+import * as yargs from 'yargs';
+
+interface TasksMap {
+  [projectName: string]: { [targetName: string]: Task };
+}
+
+type RunArgs = yargs.Arguments & ReporterArgs;
+
+export function runCommand<T extends RunArgs>(
+  projects: ProjectNode[],
```

Comment:
I'd rename it to indicate that this isn't list of all possible projects but only the projects are running tasks against this time.